pkgcore/operations/format.py

```diff
@@ -45,10 +45,11 @@ def fetch_one(self, fetchable, observer):
         # XXX hack atm, could use better logic but works for now
         try:
             fp = self.fetcher(fetchable)
-        except Exception:
+        except Exception as e:
+            observer.error(str(e))
+            observer.error("failed fetching %s", fetchable)
             fp = None
         if fp is None:
-            self.failed_fetch(fetchable, observer)
```

Comment:
This line can't be dropped, otherwise the pkg_nofetch phase doesn't get run properly on failures. See man 5 ebuild or PMS (app-doc/pms) for pkg_nofetch info.